src/nelder_mead.jl

```diff
@@ -0,0 +1,131 @@
+export nelder_mead
+
+"""
+    nelder_mead(nlp)
+
+This implementation follows the algorithm described in chapter 9 of [1].
+[1] Numerical Optimization (Jorge Nocedal and Stephen J. Wright), Springer, 2006.
+"""
+function nelder_mead(nlp :: AbstractNLPModel;
+                     x :: AbstractVector = copy(nlp.meta.x0),
+                     vertices :: Vector{<: AbstractVector{<: AbstractFloat}} = Vector{eltype(x)}[],
+                     tol :: Real = √eps(eltype(x)),
+                     max_time :: Float64 = 30.0,
+                     max_eval :: Int = -1,
+                     ref :: Real = - one(eltype(x)),
+                     exp :: Real = - one(eltype(x)) * 2,
+                     ocn :: Real = - one(eltype(x)) / 2,
+                     icn :: Real = one(eltype(x)) / 2,
+                     max_elm :: Int = 0,
+                     min_elm :: Int = 0)
```

Comment:
I think it would be better if we don't have memoization by default and create extra functions like
```julia
function nelder_mead_memo(nlp, args...; min_elm = 0, max_elm = 0, kwrags...)
  return nelder_mead(MemoNLP(nlp, min_elm, max_elm), args...; kwargs...)
end
```

Reply:
Absolutely, i think it looks a lot cleaner.